test/controllers/users_controller_test.rb

```diff
@@ -272,7 +272,25 @@ class UsersControllerTest < ActionController::TestCase
     @request.session.clear
     @request.env['HTTP_REMOTE_USER'] = 'ares'
     get :extlogin
-    assert_redirected_to edit_user_path(User.unscoped.find_by_login('ares'))
+    assert_redirected_to ApplicationHelper.current_hosts_path
+  end
+
+  test "external user with email enabled but not set should be redirected to the edit page with warning" do
```

Comment:
That is true, however; creation of external user account upon first login is automatic and as far as I know, it is not possible to make it so that mail is enabled right away. At the same time, it is not possible to save an existing user account with mail both enabled and blank since it won't pass the validation. The only way I was able to bypass this is by using the stubbed attributes, but as you say, this is already covered by the other test. So I can either leave this test as it is or I can delete it. Which do you think would be better?

Reply:
What about using fixtures or FactoryBot?

Reply:
Thanks, that worked.